router/router_test.go

```diff
@@ -130,6 +136,65 @@ func TestRouter(t *testing.T) {
 			})
 		})
 
+		Convey("When a legacy dataset request is made, but the dataset handler is not enabled", func() {
+
+			url := "/economy/inflationandpriceindices/datasets/consumerpriceinflation/current"
+			req := httptest.NewRequest("GET", url, nil)
+			res := httptest.NewRecorder()
+
+			config.DatasetEnabled = false
+			router := router.New(config)
+			router.ServeHTTP(res, req)
+
+			Convey("Then a request is sent to Zebedee to check the page type", func() {
+				So(len(zebedeeClient.GetWithHeadersCalls()), ShouldEqual, 1)
+			})
+
+			Convey("Then no request is sent to the dataset handler", func() {
+				So(len(datasetHandler.ServeHTTPCalls()), ShouldEqual, 0)
```

Comment:
@CarlHembrough Does datasetHandler need to be changed to prefixDatasetHandler as that's the thing we are checking to make sure it's not called right?

I assume the test passed because the regular datasetHandler ideally doesn't get called either?

Reply:
Yes that's right - it should be changed to prefixDatasetHandler 👍